Interaction between Attesting and Target Environment (#22)
lamps-wg/csr-attestation: draft-ounsworth-csr-attestation.md

# Security Considerations

> A certification request consists of a distinguished name, a public key,

Reviewer: #21 now uses "Certificate Signing Requests". Maybe:

```suggestion
A certificate signing request consists of a distinguished name, a public key,
```

Author: PKCS#10 defines a message and it is called "Certification Request". This message has later been called "Certificate Signing Request".

In this paragraph I am referring to PKCS#10 -- maybe I should include a reference.

Author: Proposal is here: https://github.com/lamps-wg/csr-attestation/pull/27

and optionally a set of attributes, collectively signed by the entity
requesting certification. The private key used to sign the CSR MUST

> be different from the key used to sign evidence by the attester. To

Reviewer 2: I'll retract this; I think it's a minor quibble.

# Security Considerations

A PKCS#10 or CRMF Certification Request message consists of a
distinguished name, a public key, and optionally a set of attributes,
collectively signed by the entity requesting certification.

> The private key used to sign the CSR MUST

Author: I am not aware of such a requirement. Also, you are co-authors of RFCs, which did the same.

> The private key used to sign the CSR MUST

Reviewer: Well. RFC 9124 is like the odd one standing out (the SecConSec there is *the* relevant Section containing all the requirements next to the information elements themselves). I'd not use that one as an guideline.

> be different from the Attesting Environment key used to sign Evidence about the Target Environment, which controls the private key used to sign the CSR. To

Reviewer: Are there cases in DICE where it could end up being the same key material?

Author: Yes, you could use the same key -- not just in DICE. From a key separation point of view that's not a good idea.

theft or misuse (including being non-exportable / non-recoverable),
the attesting environment has to collect claims about this secure
environment (or target environment, as shown in {{fig-attester}}).

{{fig-attester}} shows the interaction inside an attester. The
attesting environment, which is provisioned with an attestation key,
retrieves claims about the target environment. The target
environment offers key generation, storage and usage, which it
makes available to services. The attesting environment collects
these claims about the target environment and signs them and
exports evidence for use in remote attestation via a CSR.

~~~
                   ^
                   |CSR with
                   |Evidence
     .-------------+-------------.
     |                           |
     |       CSR Library         |<-----+
     |                           |      |
     '---------------------------'      |
            |  ^         ^              |
 Private    |  | Public  | Signature    |
 Key        |  | Key     | Operation    |
 Generation |  | Export  |              |
            |  |         |              |
 .----------|--|---------|------------. |

>  |          |  |         |    Attester| |

Reviewer: The text above implies that the key-material for signing the CSR is in a target environment. Following that thought it would also imply that the signing code exists next to the key material in the same target environment.

>  |          |  |         |    Attester| |

Author: It is, of course, an implementation decision of where to put what code. In the drawing I assume that the CSR library is outside the attester but the attester performs a software measurement of that code.

I will add a note to the draft.

 |          v  |         v            | |
 |    .-----------------------.       | |
 |    | Target Environment    |       | |
 |    | (with key generation, |       | |
 |    | storage and usage)    |       | |
 |    '--------------+--------'       | |
 |                   |                | |
 |           Collect |                | |
 |            Claims |                | |
 |                   |                | |
 |                   v                | |
 |             .-------------.        | |
 |Attestation  | Attesting   |        | |
 |   Key ----->| Environment +----------+
 |             |             |Evidence|

>  |             '-------------'        |

Reviewer 3: ```suggestion
 |             '-------------' abt. TE|
```

Author: I don't know what "abt. TE" means

Reviewer 2: "About Target Environment"

~~~
{: #fig-attester title="Interaction between Attesting and Target Environment"}

{{fig-attester}} places the CSR library outside the Attester, which
is an implementation choice. The CSR library may also be located
inside the trusted computing base. Regardless of the placement
of the CSR library an Attesting Environment MUST be able to collect
claims about the Target Environment such that statements about
the storage of the keying material can be made. For example, one
implementation may perform a software measurement of the CSR library
along with the crypto library implementation that has access to the

> keying material. For the Verifier the provided Evidence must allow

Reviewer 3: ```suggestion
keying material. For the Verifier, the provided Evidence must allow
```

an assessment to be made whether the key used to sign the CSR
is stored in a secure location and cannot be exported.

Evidence communicated in the attributes and structures defined
in this document are meant to be used in a CSR. It is up to
the Verifier and to the Relying Party (RA/CA) to place as much or
as little trust in this information as dictated by policies.

This document defines the transport of Evidence of different formats
in a CSR. Some of these encoding formats are based on standards
while others are proprietary formats. A Verifier will need to understand
these formats for matching the received claim values against policies.

Policies drive the processing of Evidence at the Verifier and other
policies influence the decision making at the Relying Party when
evaluating the Attestation Result. The Relying Party is ultimately
responsible for making a decision of what information in the Attestation
Result it will accept. The presence of the attributes
defined in this specification provide the Relying Party with additional

> assurance about Attester. Policies used at the Verifier and the Relying

Reviewer 3: ```suggestion
assurance about an Attester. Policies used at the Verifier and the Relying
```
